### PR TITLE
Decorators

### PR DESCRIPTION
**Pull request by AminaBuhari**
In this project, I used the Decorator design pattern to correct the names of people.
Interface

### CODE REVIEW

File: ruby_project/classes/nameable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError
```
Comment (Mre55) [inline ruby_project/classes/nameable.rb]: Kindly add a more descriptive message when an error raises. Consider doing something like this:
```
raise NotImplementedError, "#{self.class} has not implemented method '#{__method__}'"
```
Or you can customize it to your preference.

File: ruby_project/classes/trimmer_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+require_relative './decorator'
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name[0..9]
+  end
```
Comment (Mre55) [inline ruby_project/classes/trimmer_decorator.rb]: Here first this method should check whether the name is more than 10 characters or not. For that, you can use [Ternary operator ](https://syntaxdb.com/ref/ruby/ternary#:~:text=The%20ternary%20operator%20is%20used,behaving%20similar%20to%20a%20function.) and check the ``@nameable.correct_name. length > 10`` then if it's more than 10 characters return what you have now or if it's not more than 10 characters you can return it self.

**Review by Mre55 — CHANGES_REQUESTED** [review]
**Hi @AminaBuhari,**

You have done a great job so far! 💪 👏 
You still need to work on a few issues before moving on to the next project, but you're almost there!

## To highlight 🎯

- The Pull Request is following the GitHub Flow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 

## Required Changes ♻️🛠️ 
_Check the comments under the review._




## Optional suggestions 🟡 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: ruby_project/classes/trimmer_decorator.rb (hunk 2 above)
Comment (AminaBuhari) [reply ruby_project/classes/trimmer_decorator.rb]: True. Thanks for the suggestion

**Review by KDlamini — APPROVED** [review]
### Hi @AminaBuhari 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

# Status: Approved ✔️ 

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**